### PR TITLE
Add custom converter for handling empty string / NaN / other unparseable numeric values.

### PR DESCRIPTION
**Pull request by keithhuster**
* This should fix the error we are seeing in TeamCity when reports contain empty string values for float / double types.

### CODE REVIEW

File: Cake.BenchmarkDotNet/Dto/Percentiles.cs [hunk 1]
```diff
@@ -1,19 +1,15 @@
...
-        public double P0 { get; set; }
-        public double P25 { get; set; }
-        public double P50 { get; set; }
-        public double P67 { get; set; }
-        public double P80 { get; set; }
-        public double P85 { get; set; }
-        public double P90 { get; set; }
-        public double P95 { get; set; }
-        public double P100 { get; set; }
+        public long P0 { get; set; }
```
Comment (rfleming71) [inline Cake.BenchmarkDotNet/Dto/Percentiles.cs]: Looking at the sample report you uploaded, these values should be double still 

ex: `341.33`

Ref benchmark code: https://github.com/dotnet/BenchmarkDotNet/blob/63e28c100a42a6492d09a0b93a8a4c141061bb0d/src/BenchmarkDotNet/Mathematics/Statistics.cs

File: Cake.BenchmarkDotNet/Dto/ConfidenceInterval.cs [hunk 2]
```diff
@@ -3,11 +3,11 @@
     public class ConfidenceInterval
     {
         public int N { get; set; }
-        public double Mean { get; set; }
-        public double StandardError { get; set; }
+        public long Mean { get; set; }
```
Comment (rfleming71) [inline Cake.BenchmarkDotNet/Dto/ConfidenceInterval.cs]: Looking at the report you uploaded, these should still be doubles.

**Review by rfleming71 — CHANGES_REQUESTED** [review]

File: Cake.BenchmarkDotNet/BenchmarkDotNetAliases.cs [hunk 3]
```diff
@@ -10,6 +7,8 @@
 using Cake.Core.Annotations;
 using Cake.Core.Diagnostics;
 using Perfolizer.Mathematics.Thresholds;
+using System;
```
Comment (rfleming71) [inline Cake.BenchmarkDotNet/BenchmarkDotNetAliases.cs]: Please move the system usings to the top of the file.

File: Cake.BenchmarkDotNet/Comparer.cs [hunk 4]
```diff
@@ -1,21 +1,29 @@
-﻿using System;
+﻿using Cake.BenchmarkDotNet.Dto;
+using Newtonsoft.Json;
+using Perfolizer.Mathematics.SignificanceTesting;
+using Perfolizer.Mathematics.Thresholds;
+using System;
 using System.Collections.Generic;
```
Comment (rfleming71) [inline Cake.BenchmarkDotNet/Comparer.cs]: Please move the system usings to the top of the file.

File: Cake.BenchmarkDotNet/NotANumberNumericConverter.cs [hunk 5]
```diff
@@ -0,0 +1,70 @@
+﻿using Newtonsoft.Json;
+using System;
```
Comment (rfleming71) [inline Cake.BenchmarkDotNet/NotANumberNumericConverter.cs]: Please move the system usings to the top of the file.

**Review by rfleming71 — CHANGES_REQUESTED** [review]

File: Cake.BenchmarkDotNet/Dto/Percentiles.cs (hunk 1 above)
Comment (keithhuster) [reply Cake.BenchmarkDotNet/Dto/Percentiles.cs]: Fixed

File: Cake.BenchmarkDotNet/Dto/ConfidenceInterval.cs (hunk 2 above)
Comment (keithhuster) [reply Cake.BenchmarkDotNet/Dto/ConfidenceInterval.cs]: Fixed

File: Cake.BenchmarkDotNet/BenchmarkDotNetAliases.cs (hunk 3 above)
Comment (keithhuster) [reply Cake.BenchmarkDotNet/BenchmarkDotNetAliases.cs]: Fixed

File: Cake.BenchmarkDotNet/Comparer.cs (hunk 4 above)
Comment (keithhuster) [reply Cake.BenchmarkDotNet/Comparer.cs]: Fixed

File: Cake.BenchmarkDotNet/NotANumberNumericConverter.cs (hunk 5 above)
Comment (keithhuster) [reply Cake.BenchmarkDotNet/NotANumberNumericConverter.cs]: Fixed

**Review by rfleming71 — APPROVED** [review]

**Pull request merged**